corgi/core/migrations/0081_clean_cdn_relations.py

```diff
@@ -0,0 +1,56 @@
+from django.db import migrations
+
+# Non-historical model, i.e. if you change save_product_taxonomy() tomorrow
+# the migration will not run the same code as before (at the time the migration was written)
+from corgi.tasks.prod_defs import (
+    slow_remove_product_from_build as current_remove_product_from_build,
+)
+
+
+def clean_cdn_repo_relations(apps, schema_editor):
+    ProductComponentRelation = apps.get_model("core", "ProductComponentRelation")
+    ProductVariant = apps.get_model("core", "ProductVariant")
+    # Hardcoding the constant because we want to remove this type from the model
+
+    # Prefetch of CDN_REPO relation product_ref variant primary keys to make the following loop more
+    # efficient
+    cdn_relation_variants = (
+        ProductComponentRelation.objects.filter(type="CDN_REPO", software_build_id__isnull=False)
+        .values_list("product_ref", flat=True)
+        .distinct()
+    )
+
+    variant_names_pk = ProductVariant.objects.filter(name__in=cdn_relation_variants).values(
+        "name", "pk"
+    )
+    variant_names_dict = {npk["name"]: str(npk["pk"]) for npk in variant_names_pk}
+
+    for cdn_relation_build_to_product in (
+        ProductComponentRelation.objects.filter(type="CDN_REPO", software_build_id__isnull=False)
+        .values_list("software_build__pk", "product_ref")
```

Comment:
You can do `software_build_id` here, which just reuses the build's PK that's already stored in the relations table instead of having to look up the related model and get the PK from the SoftwareBuild table.
```suggestion
        .values_list("software_build_id", "product_ref")
```
Otherwise this migration looks good. Thank you so much for finding a way to clean up these old CDN repo relations in a migration / script without having to reload all the data!

Reply:
updated, thanks.